microprofile/WFLY-12342_server_readiness_probes.adoc

```diff
@@ -0,0 +1,95 @@
+= [WFLY-12342] Integrate server probes in MP Health readiness check
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,health,openshift
+
+== Overview
+
+MicroProfile Health 2.0 introduced a readiness health check.
+WFLY-12228 added support for MicroProfile Health 2.0 without enhancing the readiness health check.
+
+This means that the readiness probe always return UP by default (as long as the microprofile-health-smallrye is installed).
+
+This RFE enhances it with 3 probes (that are part of the https://github.com/jboss-container-images/jboss-eap-modules/blob/251f422c97e1bcd3625f57295bc79973193482a6/os-eap-probes/2.0/added/probes/probe/eap/dmr.py#L23[EAP readiness probe script]):
+
+* `server-status` - returns UP when the server-state is `running`
+* `boot-errors` - return UP if there are no boot-errors
+* `deployment-status` - returns UP if status for all deployments is OK
+
+Once this RFE is done, we could simplify the EAP readiness probe script by replacing its various DMR checks by a single check to `/subsystem=microprofile-health-smallrye:check-ready` (or a HTTP call to `http://:9990/health/ready`).
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12342[WFLY-12342]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1322[EAP7-1322]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[X] QE
+
+=== Affected Projects or Components
+
+* This RFE impacts the `microprofile-health-smallrye` extension provided by WildFly
+* Once done, it can be used by Kubernetes/Openshift to configure their readiness probe
```

Comment:
@jmesnil What about liveness probe on OpenShift. For example I expect bootable jar image should configure readiness probe as http call to http://:9990/health/ready. But what should be the liveness probe in this case?

Reply:
it uses the existing http call to `/health/live`: https://github.com/eclipse/jkube/blob/97571e56bb72c3e4d7f5f0aa979b872897eab336/jkube-kit/jkube-kit-wildfly-jar/src/main/java/org/eclipse/jkube/wildfly/jar/enricher/WildflyJARHealthCheckEnricher.java#L50

Reply:
Ok so you are saying in case when user is not using jkube he should configure liveness probe as `check-live`. Does `check-live` has same semantic as s2i liveness-probe.sh?

Reply:
As part of this RFE, we are not changing the liveness probes.
This RFE focuses on the readiness probe and makes sure that we have a way to check the readiness of the application server directly from HTTP (without any external scripts).

Reply:
I understand. I am thinking how bootable jar will do without python liveness probes? And eventually how wildfly can live without that python liveness probes in future. 

I understand one motivation for this RFE is provide support for bootable jar on OpenShift. But to have fully support on OpenShift we need readiness probes and liveness probes as well. Were liveness probes already discussed somewhere before? Does current capabilities  (e.g. check-live) provide enough support for bootable jar on OpenShift?

Reply:
I'm not sure to follow you.

WildFly Bootable jar is already using the http `/health/live` endpoint for the liveness probes . There is nothing else to do here. We did not need to do anything in WildFly as the server does not provide additional liveness health check procedures.

However we needed additional readiness health check procedures that are now exposed with the `/health/ready` http endpoint.

Just to clarify, the Bootable jar will rely on the *HTTP endpoints* (`/health/live` & `/health/ready`) and not on the DMR operations in the `microprofile-health-smallrye` subsystem.

Reply:
`Bootable jar will rely on the HTTP endpoints (/health/live & /health/ready) ` I was looking for that statement. Thank you.